tests/sparse/test_basic.py

```diff
@@ -1607,6 +1631,33 @@ def test_int32_dtype(self):
         a = np.asarray(np.random.randint(0, 100, (size, size)), dtype=intX)
         f(i, a)
 
+    def test_tensor_dot_types(self):
+
+        x = sparse.csc_matrix("x")
+        x_d = aet.matrix("x_d")
+        y = sparse.csc_matrix("y")
+
+        res = aet.dot(x, y)
+        op_types = set(type(n.op) for n in applys_between([x, y], [res]))
+        assert sparse.basic.StructuredDot in op_types
+        assert aet.math.Dot not in op_types
+
+        res = aet.dot(x_d, y)
+        op_types = set(type(n.op) for n in applys_between([x, y], [res]))
+        assert sparse.basic.StructuredDot in op_types
+        assert aet.math.Dot not in op_types
+
+        res = aet.dot(x, x_d)
+        op_types = set(type(n.op) for n in applys_between([x, y], [res]))
+        assert sparse.basic.StructuredDot in op_types
+        assert aet.math.Dot not in op_types
+
+        res = aet.dot(aet.second(1, x), y)
+        op_types = set(type(n.op) for n in applys_between([x, y], [res]))
+        assert sparse.basic.StructuredDot in op_types
+        assert aet.math.Dot not in op_types
+
+    # @aesara.config.change_flags(compute_test_value="raise", optimizer_verbose=True)
```

Comment:
Looks like this `change_flags` is not needed?

Reply:
No, it's not; that was left over from debugging.